core/consensus/aura/src/lib.rs

```diff
@@ -411,16 +414,16 @@ fn check_header<B: Block>(slot_now: u64, mut header: B::Header, hash: B::Hash, a
 		// check the signature is valid under the expected authority and
 		// chain state.
 
-		let expected_author = match slot_author(slot_num, &authorities) {
+		let expected_author: AuthorityId<P> = match slot_author::<P>(slot_num, &authorities) {
 			None => return Err("Slot Author not found".to_string()),
 			Some(author) => author
 		};
 
 		let pre_hash = header.hash();
 		let to_sign = (slot_num, pre_hash).encode();
-		let public = ed25519::Public(expected_author.0);
+		let public = expected_author;
 
-		if ed25519::Pair::verify(&sig, &to_sign[..], public) {
+		if <P as Pair>::verify(&sig, &to_sign[..], public) {
```

Comment:
same here just P::verify should work, but I'm ok let it explicit

Reply:
i'd prefer P::verify